torchvision/prototype/datapoints/_dataset_wrapper.py

```diff
@@ -0,0 +1,250 @@
+# type: ignore
+
+from __future__ import annotations
+
+import contextlib
+
+import functools
+from collections import defaultdict
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+import PIL.Image
+import torch
+from torch.utils.data import Dataset
+from torchvision import datasets
+from torchvision.prototype import datapoints
+from torchvision.prototype.transforms import functional as F
+
+__all__ = ["wrap_dataset_for_transforms_v2"]
```

Comment:
Curious where we're planning to expose this? I guess it'll be in `torchvision.transforms.v2` like all the transforms? In which case the `for_transforms_v2` might be slightly redundant (although I tend to prefer long descriptive names myself...)

Reply:
See https://github.com/pytorch/vision/pull/6663#discussion_r982550714. At the time of writing it was not clear if datasets v2 was going to come or not. Since this is clearer now, I'm also happy to put it under transforms. My concern here was that this wrapper links the two API's and we might get cyclic dependencies and so I put it into the place that both `datasets` and `transforms` can safely depend on.